src/widgets/entities/widget.entity.ts

```diff
@@ -4,13 +4,15 @@ import {
   Entity,
   ManyToOne,
   PrimaryGeneratedColumn,
+  Check,
 } from 'typeorm';
 import { ApiProperty } from '@nestjs/swagger';
 import { User } from '../../users/entities/user.entity';
 
 export type WidgetId = string;
 
 @Entity('widget')
+@Check('"preset" IS NULL OR "metrics" IS NULL')
```

Comment:
added db check to make sure everything is consistent

this might highlight broken widgets when the corresponding migration is run